pyaedt/generic/grpc_plugin_dll_class.py

```diff
@@ -0,0 +1,402 @@
+from ctypes import CFUNCTYPE
+from ctypes import PyDLL
+from ctypes import c_bool
+from ctypes import c_int
+from ctypes import c_wchar_p
+from ctypes import py_object
+import os
+import re
+import types
+
+from pyaedt.generic.general_methods import GrpcApiError
+from pyaedt.generic.general_methods import _retry_ntimes
+from pyaedt.generic.general_methods import settings
+
+logger = settings.logger
+
+
+class AedtBlockObj(list):
+    def GetName(self):
+        if len(self) > 0:
+            f = self[0]
+            if isinstance(f, str):
+                d = f.split(":")
+                if len(d) == 2:
+                    if d[0] == "NAME":
+                        return d[1]
+
+    def __GetValueIdxByKey__(self, keyName):
+        for i in range(0, len(self) - 1):
+            if isinstance(self[i], str):
+                toks = self[i].split(":")
+                if len(toks) == 2:
+                    if toks[1] == "=" and toks[0] == keyName:
+                        return i + 1
+        raise GrpcApiError(keyName + " is not a key!")
+
+    def __getitem__(self, idxOrKey):
+        if isinstance(idxOrKey, str):
+            idx = self.__GetValueIdxByKey__(idxOrKey)
+            if idx != None:
+                return super().__getitem__(idx)
+        return super().__getitem__(idxOrKey)
+
+    def __setitem__(self, idxOrKey, newVal):
+        if isinstance(idxOrKey, int):
+            if idxOrKey >= 0 or idxOrKey < len(self):
+                oldItem = self.__getitem__(idxOrKey)
+                if isinstance(oldItem, str):
+                    toks = oldItem.split(":")
+                    if len(toks) == 2 and (toks[1] == "=" or toks[0] == "NAME"):
+                        raise GrpcApiError("The element is a key should not be overwritten!")
+            return super().__setitem__(idxOrKey, newVal)
+        if isinstance(idxOrKey, str):
+            idx = self.__GetValueIdxByKey__(idxOrKey)
+            if idx != None:
+                return super().__setitem__(idx, newVal)
+            raise GrpcApiError("Key not found")
```

Comment:
```suggestion
            raise GrpcApiError("Key is not found.")
```
Messages should be complete (with verb) and conclude with punctuation (period).